docs/LwM2M_Client/Nordic/Runtime_certificate.md

```diff
@@ -0,0 +1,107 @@
+# Runtime Certificate and Private Key Configuration
+
+## Introduction
+Anjay library supports secure connection between device and server. Encryption can be achieved with the use of Pre-Shared Key as it was described in [nRF9160](nRF9160.md) and [Thingy:91](Thingy91.md) pages or with certificate-based cryptography.
+
+In **Certificate Mode** an asymmetrical algorithm is used to authenticate the connection endpoints and initialize payload encryption. Appropriate certificates need to be generated for both the LwM2M Client and the LwM2M Server. Public certificates of both parties are mutually available, and each party also has access to its corresponding private key.
+
+You can read more about secure communication on [Anjay's documentation](https://avsystem.github.io/Anjay-doc/BasicClient/BC-Security.html) page.
+
+## Prerequisites
+* The nRF9160DK board with a USB cable.
+* Cloned [Anjay-zephyr-client](https://github.com/AVSystem/Anjay-zephyr-client#getting-started) repository
+* Installed **nrfjprog** from [Nordic Semiconductor page](https://www.nordicsemi.com/Products/Development-tools/nrf-command-line-tools/download)
+* Installed [OpenSSL](https://www.openssl.org/source/)
+* Installed **minicom** (for Linux), **RealTerm**, **PuTTy** (for Windows), or another serial communication program.
+* An active [Coiote IoT DM](https://eu.iot.avsystem.cloud/) user account.
+
+!!! Note
+    In this tutorial we will use the nRF9160DK board as an example.
+
+## Build and flash the device
+0. Connect the nRF9160DK board to a USB port of your machine.
+0. Go to your local **Anjay-zephyr-client/demo** directory
+0. Set manifest file to `west-nrf.yml`, and do `west update`:
+
+    ```
+    west config manifest.file west-nrf.yml
+    west update
+    ```
+
+0. Build a project with a runtime certificate and private key.
+
+    ```
+    west build -b <BOARD> -p -- -DCONFIG_ANJAY_ZEPHYR_RUNTIME_CERT_CONFIG=y
+    ```
+
+    where <BOARD\> should be replaced by the selected board from `boards/` directory.
+
+    !!! Note
+
+        The runtime certificate and private key configuration do not work with Thingy:91. Other boards should work with this command.
+
+    This feature works with nRF9160DK starting from revision v0.14.0. For this board use configuration that utilizes an external flash chip and software-based cryptography:
+
+    ```
+    west build -b nrf9160dk_nrf9160_ns@0.14.0 -p -- -DCONF_FILE=prj_extflash.conf -DOVERLAY_CONFIG="overlay_nrf_mbedtls.conf"
+    ```
+
+0. Flash the board with `west flash` command.
+
+## Generate certificate
+The certificate and private key based on the SECP256R1 curve can be provided through the shell interface in `.pem` format. To generate them open terminal in the **Anjay-zephyr-client/demo** directory and use the following commands.
```

Comment:
```suggestion
The certificate and private key based on the SECP256R1 curve can be provided through the shell interface in `.pem` format. To generate a self sign certificate and key pair open a terminal in the **Anjay-zephyr-client/demo** directory and use the following commands.
```

Reply:
Proposal rejected :cry: ?

Reply:
Also I just noticed that in `overlay_nrf_mbedtls.conf` we configure support for both EC P256 and EC P384 curves. I know that this description is based on the README file in Anjay-Zephyr-client but did anyone verified that we support  only P256? Maybe we support P384 also and could say that the user can choose one of those curves?

Reply:
sorry I thought I added this

Reply:
`self-signed` btw

Reply:
About the P384 support. Discussed offline that this can stay as is.